### ISSUE TITLE
Lauching campaign makes my tests fail everytime

### ISSUE THREAD

**Issue by Thelulucien** [cerberustesting/cerberus-source#1838]
Hi !

I launched my very first campaign today after correcting old tests.
Theses tests work perfectly fine when I launch them one at a time as you can see on the screenshot

(The name of this test is : 'order_add_create_account_gift_cb')
![order_add_create_account_gift_cb](https://user-images.githubusercontent.com/17575025/45307612-bc25af00-b51f-11e8-8429-a3cf07a7debf.png)

My problem is that when I launch a campaign, and consequently several tests at the same time, theses very same tests fail. I launch my campaign through this interface : 
![campaign_launching](https://user-images.githubusercontent.com/17575025/45307837-5554c580-b520-11e8-85d9-2c5807d1c65e.png)

I don't know why but it seems that they fail when they need to perform a click action. Below, the same test as shown above, but launched within a campaign. As you can see, the failed action is a click on a button.
![test_failed_campaign](https://user-images.githubusercontent.com/17575025/45308467-cea0e800-b521-11e8-8c9b-fbcc05270066.png)

Here, the screenshot of the Cerberus' screenshot (yes, it is a screen of a screenshot, it's quite ironic). This is very intriguing because what we can see is that the product was indeed added to the cart but cerberus considered the action as not performed.
![screenshot_from_cerberus_of_last_action](https://user-images.githubusercontent.com/17575025/45308596-19226480-b522-11e8-9df6-0a93fece15ee.png)

Maybe it is my working environment maybe not. I have no idea why it does this. Do I need to reconfigure my campaign ? Or it is possible to launch one test at a time to avoid this issue ?

Thank you for your time and your tool !

Ps : I don't know if I gave enough information so if it is needed I will be happy to give it.
Ps2 : Excuse my english, I'm a baguette eater.

**Comment by vertigo17** [comment]
Hello again baguette eater ;-)
Can it be linked with something like you perform multiple simultaneous test execution at the same time on the same site with the same customer id. 
does the website you test support to do that ? Most application don’t because they share the basket inside the same customer and running multiple test on the same customer introduce cases not supposed to happen. 
Either 1 execution at a time using pool size at application / env level. Or even better start using datalib with a list of valid customer and set property to NOTINUSE. I guess you could even do a retry on the property with NOTINUSE and property calculation should loop until it is no longuer used by any other testcase.

**Comment by Thelulucien** [comment]
Hello again vertigo17 :)

You are indeed right, the website does not support multiple tests launched with the same customer simultaneously.

Using the datalib list will be perfect but I didn't understand completely your answer.

> Or even better start using datalib with a list of valid customer and set property to NOTINUSE. I guess you could even do a retry on the property with NOTINUSE 

I read in the doc that using the same name for INTERNAL data lib forms a list from which the test can pick the user he wants, so doing a list is not a problem.
I do not understand what you call a property, do you mean a subdata ? Because, for me, a property is within a testcase and is attached to it.
Do a retry ?
I'm certainly missing something in the documentation.

> [...] and property calculation should loop until it is no longuer used by any other testcase.

Again, I'm sorry but I have no idea how to do that and I did not find how in the doc.

Thanks for your quick answer and for giving me your time !

Ps : Si possible, pourriez-vous l'expliquer en français la partie concernant les NOTINUSE ?
PsTranslate : Could you explain the NOTINUSE part in french if it possible ?

**Comment by Thelulucien** [comment]
I've been lurking in issues on the github and I found this : 
[https://github.com/cerberustesting/cerberus-source/issues/1797](url)

It appears that it was a former employee/intern of my company so the solution was disable but still in our Cerberus.
I followed your solution described in this issue and the tests are executing one at a time so it's kinda working.

It is a temporary solution, one execution at a time is not sustainable but at least I can work with my tests for the moment.

However, I'm still interested in launching severals tests at the same time with different customers, taking my Cerberus skill over the top :-p 
I will continue my lurking and report anything that can help here.

Thanks again for your work !

**Comment by vertigo17** [comment]
If you managed to create the INTERNAL datalib with the list of users you want to use and plug your tests to that datalib so that they all get the customer from that list, you only need to 1/ secure that on every testcase the name of the property for the customer is the same and 2/ put NOTINUSE In nature field of those properties. 

PS You also need to secure that the nb of customers in your list in greater than the nb or // execution you defined at application level.

**Comment by Thelulucien** [comment]
Re !

With your instructions it worked with one test execution at a time !
The customers are chosen in the list and are not duplicated.

I will describe how because the documentation is (sadly) outdated, I will propose an update tomorrow or next week. However, it will be useful to other users to see that description here I suppose.
(It is outdated on the property part, that's why I didn't understand when you talked about it)

I created, just like you said, several fake users in my data lib just like this : 
![data_library_list_of_user](https://user-images.githubusercontent.com/17575025/45500205-6c90ee80-b77e-11e8-933b-d4e8f3934fc6.png)

Then, I went to a test case where I needed to create an account and I went to the property tab and then "add property". To see and fill the "Nature" field and others, I had to click on the arrow on the right.
_Note_ : the property name **NEED** to be the same in every test case. If not, it won't work. 
_Note2_ : As said by vertigo17, the nature field must be "NOTINUSE"
![test_case_property](https://user-images.githubusercontent.com/17575025/45500548-33a54980-b77f-11e8-85a9-fc150515d16d.png)

Then it's ready to use like any other data lib but it will pick a not used data in the list.

I'm having problem to run several tests at the same time but I do not think it is Cerberus' fault but the website conception's fault. I will talk with the dev team tomorrow .

Again, thank you for your help and your precise answers !
Have a great day
